### ISSUE TITLE
Is possible that FieldNumber is not number?

### ISSUE THREAD

**Issue by wirekang** [yoheimuta/go-protoparser#70]
Why FieldNumber has string type? Can I expect strconv.AtoI always return nil error for FieldNumber?

**Comment by yoheimuta** [comment]
@wirekang No, FieldNumber must be a number, according to [the spec](https://developers.google.com/protocol-buffers/docs/reference/proto3-spec#fields).

Practically, [Go's int type](https://go.dev/ref/spec#Integer_literals) can cover [the sensible FieldNumber value](https://developers.google.com/protocol-buffers/docs/reference/proto2-spec#integer_literals).

However, each size limit is different. Let's say users give 170141183460469231731687303715884105727 as FieldNumber.
`strconv.Atoi` returns an error, but it's legit in Protocol Buffer.

- ref. https://go.dev/play/p/4d4ybhrwbRk
